app/controllers/search_controller.rb

```diff
@@ -0,0 +1,28 @@
+class SearchController < ApplicationController
+  before_action :authenticate_user!
+
+  def home
+  end
+
+  # Public: Finds files from BruseFile or Tag model.
+  #
+  # query  - search query
+  #
+  # Examples
+  #
+  #   find("teesst")
+  #   # => { files: [{name: "test"}] }
+  #
+  # Returns an array of files
+  def find
+    query = params[:query]
+    # Initiates array here since ruby is a pass by reference language
+    @results = []
+    @results = BruseFile.find_by_fuzzy_name(query)
```

Comment:
Shouldn't we make sure that the searched-through files belongs onlo to the current user? Or is this happening behind the scenes somewhere that I am missing?